support-frontend/assets/pages/paper-subscription-checkout/components/summary.jsx

```diff
@@ -0,0 +1,230 @@
+            ])
+        }
+      </dl>
+    }
+  </div>
+);
+
+
+const PromotionDiscount = (props: { promotion: ?Promotion }) => (
+  <span>
+    {props.promotion && hasDiscount(props.promotion) &&
+      <div className={styles.promo}>
+        <strong className={styles.promoTitle}>
+          {props.promotion.description}
+        </strong>
+        {' '}({props.promotion.promoCode})
+      </div>
+    }
+  </span>
+);
+
+
+const ChangeSubscription = (props: { route: string }) => (
+  <a className={styles.changeSub} href={props.route}>Change Subscription</a>
+);
+
+
+const DropDownButton = (props: { onClick: Function, showDropDown: boolean }) => (
+  <button
+    aria-hidden="true"
+    className={styles.dropDown}
+    onClick={props.onClick}
+  >
+    <span className={styles.spaceRight}>{props.showDropDown ? 'Hide details' : 'Show all details'}</span>
+    <SvgDropdownArrowUp className={props.showDropDown ? styles.openState : styles.defaultState} />
+  </button>
+);
+
+
+const TabletAndDesktop = (props: {
+  billingPeriod: BillingPeriod,
+  changeSubscription: string,
+  dataList: DataListItem[],
+  description: ?string,
+  image: $Call<GridImageType, GridImg>,
+  productPrice: Price,
+  promotion: ?Promotion,
+  title: string,
+}) => (
+  <span className={styles.tabletAndDesktop}>
+    <div className={styles.img}>
+      {props.image}
+    </div>
+    <div className={styles.content}>
+      <h1 className={styles.header}>Order summary</h1>
+      <header>
+        <h2 className={styles.title} title={`your subscription is ${props.title}`}>{props.title}</h2>
+        {props.description &&
+          <h3 className={styles.titleDescription}>{props.description}</h3>
+        }
+      </header>
+      <div>
+        <PriceLabel
+          className={styles.pricing}
+          productPrice={props.productPrice}
+          promotion={props.promotion}
+          billingPeriod={props.billingPeriod}
+        />
+        <PromotionDiscount promotion={props.promotion} />
+        <DataList dataList={props.dataList} />
+      </div>
+      <ChangeSubscription route={props.changeSubscription} />
+    </div>
+  </span>
+);
+
+
+const HideDropDown = (props: {
+  billingPeriod: BillingPeriod,
+  onClick: Function,
+  productPrice: Price,
+  promotion: ?Promotion,
+  showDropDown: boolean,
+  title: string,
+}) => (
+  <div className={styles.content}>
+    <h1 className={styles.header}>Order summary</h1>
+    <header>
+      <h2 className={styles.title} title={`your subscription is ${props.title}`}>{props.title}</h2>
+    </header>
+    <DropDownButton showDropDown={props.showDropDown} onClick={props.onClick} />
+    <div>
+      <PriceLabel
+        className={styles.pricing}
+        productPrice={props.productPrice}
+        promotion={props.promotion}
+        billingPeriod={props.billingPeriod}
+      />
+      <span className={styles.pricing}>
+        &nbsp;&ndash; Voucher booklet
+      </span>
+    </div>
+  </div>
+);
+
+
+const ShowDropDown = (props: {
```

Comment:
This type could possibly be 
```
props: {
  ...PropTypes,
  deliveryMethod: string,
  onClick: Function,
  showDropDown: boolean,
}
```
Up to you which you prefer though

Reply:
Thanks!